inference-engine/src/vpu/graph_transformer/src/stages/roi_align.cpp

```diff
@@ -104,6 +110,17 @@ void FrontEnd::parseROIAlign(const Model& model, const ie::CNNLayerPtr& layer, c
     stage->attrs().set<int>(s_pooled_h, layer->GetParamAsInt("pooled_h"));
     stage->attrs().set<int>(s_sampling_ratio, layer->GetParamAsInt("sampling_ratio"));
     stage->attrs().set<float>(s_spatial_scale, layer->GetParamAsFloat("spatial_scale"));
+
+    const int width = inputs[0]->desc().dim(Dim::W);
+    const int height = inputs[0]->desc().dim(Dim::H);
+    const int channels = inputs[0]->desc().dim(Dim::C);
+    const int buffer_size = sizeof(int16_t) * width * height * channels;
+    const bool use_buffer = (width >= 200) && (mode == "avg");
```

Comment:
I see it differs from optimization version I used for estimations. Does check on mode affect Faster R-CNN performance?

Reply:
no, it does not affect performance. this optimization is implemented for "avg" case only. For "max", buffer is not needed.

Reply:
Thanks for clarification!